components/js_lang/js_lang_core.cc

```diff
@@ -569,15 +573,42 @@ bool Js_sp::execute() {
     }
   }
 
-  // Call the wrapper function passing parameters to it.
+  /*
+    We about to execute JS code, which can take time. Make this execution
+    KILLable by setting killable_isolate flag, which indicates that one
+    needs to call Isolate::TerminateExecution() to abort execution of JS
+    code in the connection as well indicates Isolate for which this method
+    needs to be called.
+  */
+  js_thd->set_killable_isolate(isolate);
```

Comment:
Shouldn't we check `killed_earlier` earlier, before calling `set_killable_isolate()`?